### ISSUE TITLE
Obtain a mesh from a texture

### ISSUE THREAD

**Issue by Perissinotto** [gradientspace/geometry3Sharp#54]
Hi,

First I explain my problem.
I have a .png binary image.
![iminputblender](https://user-images.githubusercontent.com/26706347/41418948-2c81cba4-6ff1-11e8-9ef6-c18eb57d997e.png)

I would like to create a mesh out of it where x and y is defined by the image and the depth is costant. What I have achieved is creating a mesh where every pixel is a little cube of constant depth.
This lead to a very huge ammount of cube, so I tried to use the reducer to diminish the number of triangles/vertex. Unfortunately the result is not what I expected since some cube are removed leaving some blank space betweeen the cube. 

So my question is: there is a better approach to this problem?
I found some extrude function that could help me but I don't know how to pass the .png point to it.
Do you have some suggestion?

Thanks for your time
Andrea

**Comment by rms80** [comment]
Yes this is what I was going to recommend. Create a DenseGrid3f that has the same X/Y resolution as your image, and is, say, 10 voxels high in Z. Initialize to 1.  Then for Z-layers between, say, 2 and 8, set the voxel to -1 if the pixel (x,y) is white. Pass this into DenseGridTrilinearImplicit constructor, which you can then pass into Marching Cubes to get a mesh. More details in these tutorials:

http://www.gradientspace.com/tutorials/2017/11/21/signed-distance-fields-tutorial
http://www.gradientspace.com/tutorials/2018/2/20/implicit-surface-modeling

This will smooth things out a little.  Alternately for a blocky mesh you can use Bitmap3 and VoxelSurfaceGenerator:
http://www.gradientspace.com/tutorials/2017/12/14/3d-bitmaps-and-minecraft-meshes

reduce triangle count afterwards with Reducer:
http://www.gradientspace.com/tutorials/2017/8/30/mesh-simplification

**Comment by Perissinotto** [comment]
This is EXACTLY what i was looking for. Thank you very much for your help .(both of you) :)

**Comment by Evil-Spirit** [comment]
Simply I am going to make image-to-mesh online  tool. A little bit later.
